tests/app/views/test_users.py

```diff
@@ -1393,6 +1408,25 @@ def test_400_response_if_framework_is_coming(self):
         response = self.client.get('/users/export/{}'.format(self.framework_slug))
         assert response.status_code == 400
 
+    def test_response_agreed_contract_variation(self):
+        self._setup()
+        self._put_complete_declaration()
+        self._post_complete_draft_service()
+        self._post_result(True)
+        self._post_framework_agreement()
+        self._set_framework_variation()
+        self._put_variation_agreement()
```

Comment:
I was under the impression we generally try to set up existing database state in these tests through loading things directly into the database rather than successive api calls...?